controllers/openstackmachine_controller.go

```diff
@@ -439,6 +439,11 @@ func (r *OpenStackMachineReconciler) getOrCreate(logger logr.Logger, cluster *cl
 	}
 
 	if instanceStatus == nil {
+		if openStackMachine.Spec.InstanceID != nil {
+			logger.Info("Not reconciling machine in failed state. The previously existing OpenStack instance is no longer available")
+			conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InstanceNotFoundReason, clusterv1.ConditionSeverityError, "virtual machine no longer exists")
```

Comment:
Please can you also call `openStackMachine.SetFailure()`? Deletion of the VM is one of the few cases we need to treat as unrecoverable.